### PR TITLE
force sequential load of backend library

### PR DESCRIPTION
**Pull request by johnnychen94**
This fixes the CI in the master branch. I'm not very familiar with thread and locks, so I don't know if this is the right fix.

### CODE REVIEW

**Comment by IanButterworth** [comment]
We used to lock the load process but there was some reason we stopped doing it

**Comment by IanButterworth** [comment]
Here: https://github.com/JuliaIO/ImageIO.jl/pull/22

**Comment by IanButterworth** [comment]
How did https://github.com/JuliaIO/ImageIO.jl/pull/23 pass if master isnt? Did something change in 1.6.1?

**Comment by johnnychen94** [comment]
Actually, I also tried in Julia 1.5 (with both FileIO v1.5 and FileIO v1.8) and failed to pass the test...

**Comment by IanButterworth** [comment]
So I think what happened is that closing and opening the PR in #22 didn't update the merge commit.. so the test wasn't up to date.

File: src/ImageIO.jl [hunk 1]
```diff
@@ -9,24 +9,33 @@ const idTiffImages = Base.PkgId(UUID("731e570b-9d59-4bfa-96dc-6df516fadf69"), "T
 
 ## PNGs
 
+const libload_lock = [Threads.SpinLock()]
```
Comment (IanButterworth) [inline src/ImageIO.jl]:
```suggestion
const libload_lock = Threads.ReentrantLock()
```

File: src/ImageIO.jl [hunk 2]
```diff
@@ -9,24 +9,33 @@ const idTiffImages = Base.PkgId(UUID("731e570b-9d59-4bfa-96dc-6df516fadf69"), "T
...
+function lazy_load(pkgid)
+    Base.root_module_exists(pkgid) && return Base.root_module(pkgid)
+    # If not available, lock and load the library in a sequential order
+    lock(libload_lock[1]) do
```
Comment (IanButterworth) [inline src/ImageIO.jl]:
```suggestion
    lock(libload_lock) do
```

**Review by IanButterworth — COMMENTED** [review]
These get it back to the old approach of locking that #22 removes

**Comment by johnnychen94** [comment]
>  closing and opening the PR in #22 didn't update the merge commit..

Yeah, it's a little bit different from our old Travis usage, in GitHub action, we need to rebase the PR on master...

**Pull request merged**